server.ts

```diff
@@ -87,47 +70,72 @@ export async function makeServer(
 	})
 
 	app.post('/files/', async (req, res) => {
-		try {
-			console.error('Getting file');
-			var loginResponse = await getLoggedInUserInfos(req);
-			if (!loginResponse.loggedIn) {
-				res.status(loginResponse.httpCode);
-				res.send(loginResponse.message);
-			} else {
-				const filePath = path.join(req.body.filePath as string);
-				const fileName = path.basename(filePath);
-				const fullFilePath = path.join(process.env.DOCUMENTS_PATH, filePath);
-				console.error('Getting file', fullFilePath);
-				res.setHeader('Content-Disposition', `attachment; filename="${fileName}"`);
-				res.sendFile(fullFilePath, (err) => {
-					if ( err ) {
-						console.error('Error sending file:', err);
-						res.status(500).send(err.message);
+		console.log('Getting file');
+		const prisma = getPrismaForUser(configFromDotEnv(), req.user);
+		const id = path.join(req.body.id as string);
+		const model = path.join(req.body.model as string);
+		const fileNameFromArgs = path.join(req.body.fileName as string);
+		const filePath = await prisma.$transaction(async (tx) => {
+
+			switch (model) {
+				case 'organism':
+					const org = await IDObfuscator.ensureDBObjectIsTheSame(id,
+						'bio_org', 'id_bio_org',
+						prisma, 'organism', getBioOrgToString);
+					return org.filePath;
+				case 'organismByFormIO':
+					const orgByFIO = await prisma.bio_org.findUnique({where: {id_bio_org: Number(id)}});
+					if (orgByFIO) {
+						return orgByFIO.filePath;
 					}
-					console.error('Getting file success', fullFilePath);
-				});
+					break;
+				case 'labHasHazardsChild':
+					const child = await IDObfuscator.ensureDBObjectIsTheSame(id,
+						'lab_has_hazards_child', 'id_lab_has_hazards_child',
+						prisma, 'hazard child', getLabHasHazardChildToString);
+					const submission = JSON.parse(child.submission);
+					return submission.data.fileLink;
+				case 'hazardAdditionalInfo':
+					const info = await IDObfuscator.ensureDBObjectIsTheSame(id,
+						'lab_has_hazards_additional_info', 'id_lab_has_hazards_additional_info',
+						prisma, 'hazard child', getLabHasHazardsAdditionalInfoToString);
+					return info.filePath;
```

Comment:
Same remark as above: don't trust client-provided file names; find another way for the client to point at what it wants.